app/test_utils.go

```diff
@@ -90,7 +90,7 @@ func newTestStdFee() auth.StdFee {
 // GenerateAddress generates an Ethereum address.
 func newTestAddrKey() (sdk.AccAddress, tmcrypto.PrivKey) {
 	privkey, _ := crypto.GenerateKey()
-	addr := ethcrypto.PubkeyToAddress(privkey.PublicKey)
+	addr := ethcrypto.PubkeyToAddress(privkey.ToECDSA().PublicKey)
```

Comment:
This is pretty ugly, can we abstract away `ToECDSA` or is it necessary?

Reply:
It is necessary because I had to change the underlying type of the secp PrivateKey to a byte slice, so the alternative of using a geth function to convert to an ecdsa private key type is worse. 

The way it was previously was bad because it assumes the type aliases the ecdsa.PrivateKey type anyway.

Reply:
mmmm yeah i just looked at the imports etc.. yeah this is good